src/lib/Microsoft.Health.Events/EventConsumers/Service/EventBatchingService.cs

```diff
@@ -144,7 +147,23 @@ private async Task UpdateCheckpoint(IEnumerable<IEventMessage> events, IEnumerab
 
         private async Task CompleteProcessing(string partitionId, IEnumerable<IEventMessage> events, string triggerReason)
         {
-            await _eventConsumerService.ConsumeEvents(events);
+            IList<IEventMessage> eventsToSubmit = new List<IEventMessage>();
```

Comment:
The idea here was to process portions of the batch in parallel across different Tasks.